File: problem_001.py
# find multiples of 3
# find multiples of 5
# exclude double-counting of things that are both
# add

mult_3 = []
mult_5 = []
doubles = []
total = 0

for x in xrange(1, 1000):
	if x % 3 == 0:
		mult_3.append(x)
	if x % 5 == 0:
		mult_5.append(x)

for x in mult_3:
	if x % 5 == 0:
		doubles.append(x)
		mult_3.remove(x)
		mult_5.remove(x)

for x in mult_3:
	total += x
for x in mult_5:
	total += x
for x in doubles:
	total += x

print total

File: problem_001.js
//If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
//
//Find the sum of all the multiples of 3 or 5 below 1000.

UPPER_LIMIT = 1000;
total = 0;

for (x = 0; x < UPPER_LIMIT; x++) {
    if (x % 3 == 0) {
        total += x;
    } else if (x % 5 == 0) {
        total += x;
    }
}

console.log(total);